packages/core/src/components/discord-messages/DiscordMessages.ts

```diff
@@ -95,7 +139,34 @@ export class DiscordMessages extends LitElement implements LightTheme {
 	}
 
 	protected override render() {
-		return html`<slot></slot>`;
+		let channelIcon: ReturnType<typeof html>;
+		switch (this.type) {
+			case 'channel':
+				channelIcon = html`${ChannelIcon()}`;
+				break;
+			case 'voice':
+				channelIcon = html`${VoiceChannel()}`;
+				break;
+			case 'locked':
+				channelIcon = html`${LockedVoiceChannel()}`;
+				break;
+			case 'thread':
+				channelIcon = html`${ChannelThread()}`;
+				break;
+			case 'forum':
+				channelIcon = html`${ChannelForum()}`;
+				break;
+		}
+
+		return html`<div>
+			${this.type && this.name
+				? html`<div class="discord-channel-header">
+						<div class="discord-channel-icon">${channelIcon}</div>
+						<div class="discord-channel-name">${this.name}</div>
+					</div>`
+				: ''}
+			<slot></slot>
+		</div>`;
```

Comment:
If type and name are not defined this should probably just return `null` so no empty `div` is rendered. If that breaks anything and the empty div is required then use the `when` directive instead of this ternary operator that has an empty string as the falsey value, similar to

https://github.com/skyra-project/discord-components/blob/0453ac02e11f29996af28579660dc0bde47f775d/packages/core/src/components/discord-embed/DiscordEmbed.ts#L328